corehq/apps/accounting/invoicing.py

```diff
@@ -518,14 +518,18 @@ def unit_cost(self):
 
         sms_count = 0
         for billable in self.sms_billables:
-            sms_count += 1
+            sms_count += billable.multipart_count
             if sms_count <= self.rate.monthly_limit:
                 # don't count fees until the free monthly limit is exceeded
                 continue
-            if billable.usage_fee:
-                total_excess += billable.usage_fee.amount
-            if billable.gateway_fee:
-                total_excess += billable.gateway_charge
+            else:
+                total_message_charge = billable.gateway_charge + billable.usage_charge
+                num_parts_over_limit = sms_count - self.rate.monthly_limit
+                already_over_limit = num_parts_over_limit >= billable.multipart_count
+                if already_over_limit:
+                    total_excess += total_message_charge
+                else:
+                    total_excess += total_message_charge * num_parts_over_limit / billable.multipart_count
```

Comment:
> Do you consider it problematic importing anything from \_\_future__

Not typically. The main argument against it is that it's not always obvious, which can be a problem if a code base does not use `__future__` imports consistently. `unicode_literals` is a little special in that it makes it impossible to create a literal byte string (and I think a few other more complicated reasons that I can't recall right now).